SG14/slot_map.h

```diff
@@ -132,24 +128,20 @@ class slot_map
     // If the check fails it is undefined behavior.
     // O(1) time and space complexity.
     //
-    constexpr reference operator[](const key_type& key)              { return *find_unchecked(key); }
-    constexpr const_reference operator[](const key_type& key) const  { return *find_unchecked(key); }
+    constexpr reference operator[](const key_type& key) {
+        return const_cast<reference>(static_cast<const slot_map*>(this)->operator [](key));
```

Comment:
This seems to replace a simple expression with a crazy complicated one. I hope on second glance you'd agree that the old code was actually more DRY? ;)
Irrelevant nitpick: in general I'd write `(*this)[key]` where you wrote `this->operator[](key)`.

Reply:
Oops. Will remove this one. I agree it is extreme for one liners :)